ingestion_server/ingestion_server/cleanup.py

```diff
@@ -272,24 +272,72 @@ def _clean_data_worker(rows, temp_table, sources_config, all_fields: list[str]):
     return cleaned_values
 
 
-def save_cleaned_data(result: dict) -> dict[str, int]:
-    log.info("Saving cleaned data...")
-    start_time = time.time()
+class CleanDataUploader:
+    # Number of records to buffer in memory before writing to disk
+    mem_buffer_size = 100
 
-    cleanup_counts = {field: len(items) for field, items in result.items()}
-    for field, cleaned_items in result.items():
-        # Skip the tag field because the file is too large and fills up the disk
-        if field == "tag":
-            continue
-        if cleaned_items:
-            with open(f"{field}.tsv", "a") as f:
-                csv_writer = csv.writer(f, delimiter="\t")
-                csv_writer.writerows(cleaned_items)
+    # Size (in MB) of local file buffer before writing to S3
+    disk_buffer_size = 10 * 1024 * 1024
 
-    end_time = time.time()
-    total_time = end_time - start_time
-    log.info(f"Finished saving cleaned data in {total_time},\n{cleanup_counts}")
-    return cleanup_counts
+    buffer = {
+        field: {"part": 1, "rows": []}
+        for field in _cleanup_config["tables"]["image"]["sources"]["*"]["fields"]
+    }
```

Comment:
Love this suggestion! I didn't think of it before. It can't be a NamedTuple because I'm modifying the `part` and `rows` but a dataclass does the job perfectly! Done 💯